src/hammer-tech/stackup.py

```diff
@@ -71,41 +73,45 @@ class WidthSpacingTuple(NamedTuple('WidthSpacingTuple', [
     __slots__ = ()
 
     @staticmethod
-    def from_setting(d: dict) -> "WidthSpacingTuple":
+    def from_setting(grid_unit: Decimal, d: dict) -> "WidthSpacingTuple":
         # pylint: disable=missing-docstring
-        width_at_least = float(d["width_at_least"])
-        min_spacing = float(d["min_spacing"])
-        assert width_at_least >= 0.0
-        assert min_spacing > 0.0
+        width_at_least = coerce_to_grid(d["width_at_least"], grid_unit)
+        min_spacing = coerce_to_grid(d["min_spacing"], grid_unit)
+        assert width_at_least >= 0
+        assert min_spacing > 0
         return WidthSpacingTuple(
             width_at_least=width_at_least,
             min_spacing=min_spacing
         )
 
+
     @staticmethod
-    def from_list(l: List[dict]) -> List["WidthSpacingTuple"]:
-        out = sorted(list(map(WidthSpacingTuple.from_setting, l)), key=lambda x: x.width_at_least)
+    def from_list(grid_unit: Decimal, l: List[dict]) -> List["WidthSpacingTuple"]:
+        out = sorted(list(map(partial(WidthSpacingTuple.from_setting, grid_unit), l)), key=lambda x: x.width_at_least)
 
         # Check that spacings increase.
-        current_spacing = 0.0
+        current_spacing = Decimal(0)
         for wst in out:
             assert wst.min_spacing >= current_spacing
             current_spacing = wst.min_spacing
         return out
 
 
 class Metal(NamedTuple('Metal', [
```

Comment:
Not to nitpick on this too much: would like to see this written along something like this:

```python
class Metal(NamedTuple('Metal', [
        ('name', str),
        ('index', int),
        ('direction', RoutingDirection),
        ('min_width', Decimal),
        ('pitch', Decimal),
        ('offset', Decimal),
        ('power_strap_widths_and_spacings', List[WidthSpacingTuple]),
        # Note: grid_unit is not currently parsed as part of the Metal data structure!
        # See #379
        ('grid_unit', Decimal)
])):
```

Reply:
OK, that's fine with me, I'll make the change.

Reply:
done